sqllogictest/src/runner.rs

```diff
@@ -802,6 +802,16 @@ impl<D: AsyncDB> Runner<D> {
         self.run_multi_async(records).await
     }
 
+    /// Run a sqllogictest script with a given script name.
+    pub async fn run_script_with_name_async(
+        &mut self,
+        script: &str,
+        name: impl Into<Arc<str>>,
+    ) -> Result<(), TestError> {
+        let records = parse_with_name(script, name).expect("failed to parse sqllogictest");
+        self.run_multi_async(records).await
+    }
+
     /// Run a sqllogictest file.
     pub async fn run_file_async(&mut self, filename: impl AsRef<Path>) -> Result<(), TestError> {
         let records = parse_file(filename)?;
```

Comment:
I think you can also parse name here? Thanks for fixing! I'll run a local test soon and get this merged.

Reply:
I believe `parse_file` includes the `filename` as the `file` field in the `Location` provided to `parse_file_inner`. So we should be all good for `run_file*` functions

See:
https://github.com/risinglightdb/sqllogictest-rs/blob/f7c12f79436a8384f952f2652d43c8d2978fb4e4/sqllogictest/src/parser.rs#L593-L597